modules/ROOT/pages/clustering/glossary.adoc

```diff
@@ -0,0 +1,123 @@
+| A marker the client can request from the cluster to ensure that it is able to read its own writes so that the application's state is consistent and only databases that have a copy of the bookmark are permitted to respond.
+
+[[glossary-causal-consistency]]
+| Causal consistency
+| When a client (driver) creates a session and executes a query, the responding server issues the client a bookmark.
+This reflects the state of the database copy on that server at the time the query was executed.
+The bookmark is passed along and updated by all subsequent queries in the session, regardless of which server executes what query.
+A bookmark can only be updated monotonically increasing.
+If a server is behind the state in the bookmark, it waits until it has caught up, or time out the query.
+Thus, clients executing queries within a session are guaranteed to read their own writes, and only see successively later states of the database.
+This is sometimes also referred to as <<glossary-session-consistency,session consistency>>.
+
+[[glossary-cluster]]
+| Causal cluster
+| A collection of servers running Neo4j that are configured to communicate with each other.
+The servers can be either _Primary_ or _Secondary_, where the Primary servers allow read and write operations and the Secondary servers allow only read operations.
+The cluster guarantees that a client application can read at least its own writes.
+See also <<glossary-primary, Primary server>>, <<glossary-secondary, Secondary server>>, and <<glossary-causal-consistency, Causal consistency>>.
+
+[[glossary-database]]
+| Database
+| The data store for the nodes, relationships, and properties that make up the graph.
+Multiple databases can be hosted on a Database Management Server (DBMS).
+
+[[glossary-dbms]]
+| Database Management System (DBMS)
+| The Neo4j services and system database running on an instance of a single server or cluster to provide one or more databases.
+
+[[glossary-disaster-recovery]]
+| Disaster recovery
+| A manual intervention to restore availability of a cluster, or databases within a cluster.
+
+[[glossary-election]]
+| Election
+| In the event that a leader becomes unresponsive, followers automatically trigger an election and vote for a new leader.
+A majority is required for the vote to be successful.
+
+[[glossary-fault-tolerance]]
+| Fault tolerance
+| A guarantee that a database can maintain persistence and availability in the event of one or more failures.
+The number of failures `f` that can be tolerated is dependent on the number of primaries `n` for the database and follows the formula `f = (n-1)/2`.
+In the event that more than `f` primaries fail, the database can no longer process write transactions and becomes read-only.
+
+[[glossary-follower]]
+| Follower
+| A primary copy of a database acting as a follower, receives and acknowledges synchronous writes from the leader.
+
+[[glossary-leader]]
+| Leader
+| A single Primary server is designated as the leader.
```

Comment:
Leadership is per database. Different cores can be leaders for different databases. The 5.x glossary entry is fine here.

Reply:
But in 5.x, we say _A single primary copy of a database_, this concept doesn't really apply in 4.4, does it? Also, the last sentence in the description does express that leadership is per database.